atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableCreator.java

```diff
@@ -0,0 +1,137 @@
+
+class CassandraTableCreator {
+    private static final String ROW = "key";
+    private static final String COLUMN = "column1";
+    private static final String TIMESTAMP = "column2";
+    private static final String VALUE = "value";
+
+    private final CassandraClientPool clientPool;
+    private final CassandraKeyValueServiceConfig config;
+
+    CassandraTableCreator(CassandraClientPool clientPool, CassandraKeyValueServiceConfig config) {
+        this.clientPool = clientPool;
+        this.config = config;
+    }
+
+    void createTables(Map<TableReference, byte[]> tableRefToMetadata) {
+        try {
+            clientPool.runWithRetry(client -> {
+                for (Map.Entry<TableReference, byte[]> entry : tableRefToMetadata.entrySet()) {
+                    createTable(entry.getKey(), entry.getValue(), client);
+                }
+                CassandraKeyValueServices.waitForSchemaVersions(config, client, "after adding the column family for "
+                        + "tables " + tableRefToMetadata.keySet() + " in a call to create tables");
+                return null;
+            });
+        } catch (Exception e) {
+            throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+        }
+    }
+
+    private void createTable(TableReference tableRef, byte[] metadata, CassandraClient client) throws TException {
+        CqlQuery query = constructQuery(tableRef, metadata);
+        client.execute_cql3_query(query, Compression.NONE, org.apache.cassandra.thrift.ConsistencyLevel.QUORUM);
+    }
+
+    private CqlQuery constructQuery(TableReference tableRef, byte[] metadata) {
+        StringBuilder queryBuilder = new StringBuilder();
+
+        TableMetadata tableMetadata = CassandraKeyValueServices.getMetadataOrDefaultToGeneric(metadata);
+        boolean appendHeavyReadLight = tableMetadata.isAppendHeavyAndReadLight();
+
+        Map<String, String> settings = new HashMap<>();
+        settings.put("bloom_filter_fp_chance", falsePositive(tableMetadata.hasNegativeLookups(), appendHeavyReadLight));
+        settings.put("caching", "'KEYS_ONLY'");
+        settings.put("compaction", getCompaction(appendHeavyReadLight));
+        settings.put("compression", getCompression(tableMetadata.getExplicitCompressionBlockSizeKB()));
+        settings.put("gc_grace_seconds", Integer.toString(config.gcGraceSeconds()));
+        settings.put("id", "'%s'");
+        settings.put("populate_io_cache_on_flush", cachePriorityIsHottest(tableMetadata));
+        settings.put("speculative_retry", "'NONE'");
+
+        queryBuilder.append("CREATE TABLE IF NOT EXISTS \"%s\".\"%s\""
+                + " ( " + ROW + " blob, " + COLUMN + " blob, " + TIMESTAMP + " bigint, " + VALUE + " blob, "
+                + "PRIMARY KEY (" + ROW + ", " + COLUMN + ", " + TIMESTAMP + ")) "
+                + "WITH COMPACT STORAGE ");
+
+        settings.forEach((option, value) -> queryBuilder.append("AND " + option + " = " + value + " "));
```

Comment:
Eventually we should have a separate object that builds up these options, taking care of the valid keys as well as the formatting of the values to be in single quotes / be legitimate objects. I realise this is high priority, so not blocking.

Reply:
The datastax driver does all of this for you! And has built in schema version agreement checking!
https://docs.datastax.com/en/latest-java-driver-api/com/datastax/driver/core/schemabuilder/SchemaBuilder.html

Reply:
I get actually using the driver to do the call (and thus getting the free schema version checking) might be hard, but might at least be able to use the SchemaBuilder class to generate the `CREATE TABLE` statement here more easily.